stacks/devel.json

```diff
@@ -148,15 +148,17 @@
         "org.opencontainers.image.title": "rocker/cuda",
         "org.opencontainers.image.description": "NVIDIA CUDA libraries added to Rocker image."
       },
-      "FROM": "nvidia/cuda:11.8.0-cudnn8-devel-ubuntu24.04",
+      "FROM": "nvidia/cuda:11.8.0-cudnn8-devel-ubuntu22.04",
```

Comment:
If I remember correctly, this value is updated by the script, so don't update it here manually.

Reply:
I wondered about this when the script added it.  Are we sure a) the production cycle for Ubuntu 24.04 has started _and_ b) that NVidia already publishes to it?  I was under the impression NVidia was particularly 'laggy' ie the `gcc` version implied by `nvcc` was always ages behind.

Reply:
There is no `nvidia/cuda:11.8.0-cudnn8-devel-ubuntu24.04`. There are

* `nvidia/cuda:11.8.0-cudnn8-devel-ubuntu22.04`
* `nvidia/cuda:11.8.0-cudnn8-devel-ubuntu20.04`
* `nvidia/cuda:11.8.0-cudnn8-devel-ubuntu18.04` (EOL)

See https://gitlab.com/nvidia/container-images/cuda/-/tree/master/dist/11.8.0

Reply:
What I mean is "not read", but "don't set it manually, because if the script is execute by CI, it will be updated".

The priority here is to use the same Ubuntu version as `rocker/r-ver`. I haven't found an easy way to check if that image actually exists (This is definitely possible using buildx, but I couldn't figure out how to check without logging into the Container Registry.), so no such processing is implemented.
If you try to build it, it will simply fail.

Currently pointing to Ubuntu 24.04 is not as intended, I really wanted it to be 22.04, which is the same as the current `ubuntu:latest`. In other words, this is a bug in the script.
https://github.com/rocker-org/rocker-versioned2/blob/7d088379de5b360408539a62f5ed5548a9527cd6/stacks/devel.json#L26

https://github.com/rocker-org/rocker-versioned2/blob/7d088379de5b360408539a62f5ed5548a9527cd6/build/make-stacks.R#L449-L450

Reply:
right, so it now reads 22.04, and I think should work as intended.  let me know if anything needs to be changed?

Reply:
> right, so it now reads 22.04, and I think should work as intended. let me know if anything needs to be changed?

Run the scripts to update the files. It should be back.

https://github.com/rocker-org/rocker-versioned2/blob/7d088379de5b360408539a62f5ed5548a9527cd6/build/README.md?plain=1#L5-L12

Reply:
@eitsupi ok I see what you mean, `make-stacks.R` tries to figure this out automatically and it's coming up with the nonsense answer of 24.04 as the answer on the devel image.  I still think that's confusing (doesn't that mean we're just failing to build a devel flavor for the whole cuda stack?) but I agree it's a separate issue and not one that needs to be solved in this PR.

I've just pushed an edit to roll that back.  

I thought I fixed whitespace but am still struggling with the linter :-(